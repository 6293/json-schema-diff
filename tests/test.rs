use insta::{assert_debug_snapshot, glob};
use json_schema_diff::diff;
use serde_json::Value;

#[test]
fn test_from_fixtures() {
    let test = |path: &std::path::Path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let value: Value = serde_json::from_str(&contents).unwrap();
        let diff = diff(value["lhs"].clone(), value["rhs"].clone()).unwrap();
        assert_debug_snapshot!(diff);
    };
    glob!("../tests/fixtures", "**/*.json", test);
}
