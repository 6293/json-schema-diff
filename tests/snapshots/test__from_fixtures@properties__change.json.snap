---
source: tests/test.rs
expression: diff
input_file: tests/fixtures/properties/change.json
---
[
    Change {
        path: ".hello",
        change: TypeRemove {
            removed: String,
        },
    },
    Change {
        path: ".hello",
        change: TypeAdd {
            added: Number,
        },
    },
    Change {
        path: ".hello",
        change: TypeAdd {
            added: Integer,
        },
    },
]
